src/resource.jl

```diff
@@ -270,42 +270,58 @@ function forget_failures()
     end
 end
 
-function tarball_git_hash(tarball::String)
-    local tree_hash
-    mktempdir() do tmp_dir
-        run(`tar -C $tmp_dir -zxf $tarball`)
-        tree_hash = bytes2hex(Pkg.GitTools.tree_hash(tmp_dir))
-        chmod(tmp_dir, 0o777, recursive=true)
-    end
-    return tree_hash
-end
-
 function download(server::String, resource::String)
     @info "downloading resource" server=server resource=resource
     hash = let m = match(hash_part_re, resource)
```

Comment:
Do we ever use this to download things that we don't actually know a hash for? If not, it might be worth just simplifying all of this to avoid the complexity of support checking the hash and not checking the hash. The only thing we fetch that doesn't have a hash would be `/registries` and that could just as well use its own separate code path for fetching instead of this one since it doesn't really need to save anything to disk.

Reply:
Agreed.